crates/pop-contracts/src/call.rs

```diff
@@ -309,9 +310,13 @@ mod tests {
 		const LOCALHOST_URL: &str = "ws://127.0.0.1:9944";
 		let temp_dir = generate_smart_contract_test_environment()?;
 		mock_build_process(temp_dir.path().join("testing"))?;
-		// Run the contracts-node.
-		let cache = temp_dir.path().join("cache");
-		let process = run_contracts_node(cache, None).await?;
+
+		// temp_dir gets dropped prematurely, so manually create a tmp directory.
+		let cache = dirs::cache_dir().expect("cache_dir failed").join("pop_tmp_call_works");
```

Comment:
As above, use https://doc.rust-lang.org/std/env/fn.temp_dir.html if temp_dir doesnt work, so that stuff is at least auto-cleaned on restart. Currently the remove_dir_all will not run if there is an error in any of the below lines using ?